Fix High Contrast/Forced Colors issues

Yan-Thomas opened the pull request:
#### What kind of changes does this PR include?

- Changes to the docs site code

#### Description

- Closes #620 

This PR fixes a couple of forced colors (high contrast) mode issues we had in the docs. Here's a comparison (using Chrome Dev Tools forced colors simulation):

| PR  |  docs.astro.build  |
| ------------------- | ------------------- |
| ![image](https://user-images.githubusercontent.com/61414485/177423187-511d6ca6-5f97-45d1-b060-1ae744d47f38.png) | ![image](https://user-images.githubusercontent.com/61414485/177423219-b7b5bb03-162f-471b-ba1f-32e5c8156e50.png) |

I made one commit per component so you can easily check all the components changed that aren't part of the comparison above in the PR's commits list.

NOTES: 
- Some changes use the `forced-colors` media query. If you search it up, you can see it doesn't have major browser support yet. I opted for it instead of the `-ms-high-contrast` media query because it is now being deprecated and doesn't work in different operating systems (AFAIK).
- I tried but I didn't find a way to change the DocSearch component colors, so I think this is something Algolia has to fix on their end.
- I also couldn't change Netlify's logo colors for some strange reason, although I wasn't going to add this change right now anyway because I'm not sure if this can lead to legal problems for changing the company's brand colors.

delucis commented:
Awesome work — thank you @Yan-Thomas! Paging @mayank99 to take a look as I know they were thinking about this in #620.

Couple of things I spotted on a quick pass:

- On this page arrow (`--theme-accent-secondary`) looks like it still needs an update:
    <img width="164" alt="image" src="https://user-images.githubusercontent.com/357379/177744246-18b2f806-136a-45a5-aca7-2511c4a05c72.png">

- In light mode, the tab components don’t make it clear which item is selected (just the bold text which is a subtle difference):
    <img width="337" alt="image" src="https://user-images.githubusercontent.com/357379/177744602-6045810c-dfc3-4bb7-98b6-332373d422da.png">

A note on the colour changes: you might find it easier to set some things globally in `public/theme.css`? The variables could be updated using the high contrast media query there?

Yan-Thomas commented:
> (Should also say I’m also checking in Chrome with dev tools forced-colors, so would be curious if anyone with a Windows machine could turn on high contrast mode and see how this looks.)

Forgot to mention this, I checked all the contrast modes in Windows (there is another theme but I can't select it for some reason):

| Aquatic  |  Desert |  Night Sky |
| ------------------- | ------------------- | ------------------- |
|  ![image](https://user-images.githubusercontent.com/61414485/177792251-43d79bd3-e9e5-4424-8c3b-f90438dc001d.png) | ![image](https://user-images.githubusercontent.com/61414485/177792385-8a188ae7-0b9e-494f-9d89-81ed7b7ffeda.png)| ![image](https://user-images.githubusercontent.com/61414485/177792085-9ddd3d68-04af-4fe1-bcda-d7c2f0126192.png) |

> They are — thank you! But now the tabs in dark mode use yellow instead of cyan for the current item, which doesn’t seem very high contrast compared to the white. Maybe the current item can be cyan in both light & dark?

I think there's no way to use the same color in both modes since we are using system colors and they change depending on the OS/contrast theme, but maybe this could work?

Light:
![image](https://user-images.githubusercontent.com/61414485/177790528-d7533bd2-416f-487a-a053-f0844cd2bdf3.png)

Dark:
![image](https://user-images.githubusercontent.com/61414485/177790671-d5d7dc01-4920-43fe-947e-5bcb7bf69a61.png)

Other alternative is to just add an outline to specify the current tab, like this:
![image](https://user-images.githubusercontent.com/61414485/177790955-9bf0842f-db1f-4a3d-958b-5bf57e267bab.png)

Yan-Thomas commented on src/components/TabGroup/TabGroup.css:
```diff
@@ -16,12 +17,23 @@
+	}
+
+  .TabGroup .active {
+    border-bottom-color: Mark;
```
Thanks for the link, I only found these system colors on MDN and they didn't had proper explanations and I thought `Mark` was just some different type of `Highlight`, but yes, I agree its better to use Highlight here.

Yan-Thomas replied:
Thanks for pointing this out, I noticed this a while ago and already fixed in my local branch. It's not related to the `Mark` color, I just had to add one more line of CSS:

```css
.TabGroup {
    border-bottom-color: CanvasText;
}
```

Yan-Thomas commented on src/components/TabGroup/TabGroup.css:
```diff
@@ -16,12 +17,23 @@
 
+@media (forced-colors: active) {
+	.TabGroup a, .TabGroup button {
+		border-bottom-color: CanvasText;
```
I considered this but it feels like the inactive tab isn't a tab at all, I suggested using a outline below, let me know what you think of this.

mayank99 replied:
I have concerns about `outline` (see above). Maybe the text of tabs should use `LinkText`? That way the inactive tab would still invite clicks.

Yan-Thomas commented on public/index.css:
```diff
@@ -504,6 +504,8 @@ h2.heading {
 .current-header-link {
 	background-color: var(--theme-bg-accent);
+	/* Indicates the current heading for forced colors users, transparent is changed to a solid color */
+	outline: 1px solid transparent;
```
I used invisible `outline`s mainly because they have more support than the `forced-colors` media query and also the fact that they don't increase the element's width/height. Maybe we can go the progressive enhancement route and change the outline for a border for users with browsers that support `forced-colors`, what do you think?

Thanks for the review!

Yan-Thomas commented:
Thanks for all the feedback. I did some changes, let me know what you guys think:

- Changed the tabs styling, so the current tab is highlighted but the others have only the link text but no `border-bottom`
- Added a border for focusable elements, so if the user is in a older browser he still has the outline, but in recent browsers the border avoids the outline from disappearing when you focus:
![image](https://user-images.githubusercontent.com/61414485/177832375-9f9d6d2b-a52c-4382-bcd2-6c931f90db07.png)
- Changed the Deploy Guide selection styling so it's more subtle.
- I changed the header logo colors so they are more semantically correct.

mayank99 reviewed (approved):
GitHub won't let me resolve my own comments but LGTM! Great work @Yan-Thomas 👍💜

The site looks absolutely gorgeous with Windows 11 themes! Also want to take a moment to appreciate Netlify's PR deploys letting me test these changes in a VM in macOS without fiddling with local clones and builds. We're really living in 2022 😄

https://user-images.githubusercontent.com/9084735/177847541-5f70887e-5b67-4e9b-9096-eb0ce41c6909.mov

delucis reviewed (approved):
Amazing work @Yan-Thomas and thanks so much for the detailed review @mayank99 💜 

LGTM! 🚀

Yan-Thomas commented:
Time to merge! 🚀
